packages/block-library/src/post-comment-avatar/edit.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	InspectorControls,
+	__experimentalUseBorderProps as useBorderProps,
+	__experimentalImageSizeControl as ImageSizeControl,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { PanelBody, ResizableBox } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __, isRTL } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+export default ( { attributes, context: { commentId }, setAttributes } ) => {
+	const { className, style, height, width, alt, url } = attributes;
+
+	const { comment } = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			return {
+				comment: getEntityRecord( 'root', 'comment', commentId ),
+			};
+		},
+		[ commentId ]
+	);
+	const borderProps = useBorderProps( attributes );
+	const authorAvatarUrls = comment?.author_avatar_urls; // eslint-disable-line camelcase
+	const avatarUrls = authorAvatarUrls
+		? Object.values( authorAvatarUrls )
+		: null;
+	const [ { defaultWidth, defaultHeight }, setDefaultSize ] = useState( {} );
+
+	// Set default widht, height and url on first render.
+	useEffect( () => {
+		setDefaultSize( { defaultWidth: width, defaultHeight: height } );
```

Comment:
I haven't checked all the nuances with the avatars, but I think we should take into account the core handling. For example this filter: https://developer.wordpress.org/reference/functions/rest_get_avatar_sizes/, which is responsible for `author_avatar_urls` value.

Also if we go this direction with a default value it shouldn't be the current `width` as this can change from the control and would work as you expected on insertion only. If we end up not  getting this dynamically, we should have a hardcoded value (if it makes sense).